Automattic/wp-desktop
# Implement electron-updater

### Description
This PR replaces the built in auto updater with `electron-updater`. 

### Motivation and Context
- Enable auto update for Windows
- Simplify release process (using Github Releases for release management)
- Update manual updater to use Github Releases as well

### How to test (dev env)
1. Create `dev-app-update.yml` in your project root with the following content: 
    ```yml
    owner: automattic
    repo: wp-desktop
    provider: github
    ```
1. In `package.json` Set version to `3.5.0`
1. Execute `make dev-server` and `make dev` to launch the app
1. Wait for the updater to prompt. 
    - on confirm: macOS & Windows performs auto update
    - Linux opens Github Release page in browser. 

### How to test (production)
1. Get installer binaries from here: https://circleci.com/gh/Automattic/wp-desktop/26710#artifacts/containers/0
1. Wait for the updater to prompt. 
    - on confirm: macOS & Windows performs auto update
    - Linux opens Github Release page in browser. 

### Todos
